azure-kusto-data/tests/sample.py

```diff
@@ -52,6 +52,10 @@
 user_assigned_client_id = "the AAD identity client id"
 kcsb = KustoConnectionStringBuilder.with_aad_managed_service_identity_authentication(cluster, client_id=user_assigned_client_id)
 
+# In case you want to authenticate with Azure CLI.
+# Running "az login" is not sufficient. Need to run "az account get-access-token" to cache token locally.

```

Comment:
```suggestion
# Users are required to be in a logged in state in az-cli, for this authentication method to succeed. Run `az login` to login to azure cli.

```

Reply:
I actually checked this and az login if sufficient (commented in #285 IIRC)

Reply:
Thanks for checking. Yes, I can clear this up a bit. Tokens have an expiration time of 60 minutes max. Az cli automatically refreshes tokens when needed for internal az cli use. Given azure-kusto-python is external to az cli and wants to use a token from there, we need to trigger a refresh to guarantee access. I found this info [here](https://github.com/Azure/azure-cli/issues/6234#issuecomment-387848841).

This also follows the contributor's [comments](https://github.com/Azure/azure-kusto-python/releases/tag/v0.0.43) for this feature. 

How about if we use this? Let me know.

Users are required to be in a logged in state in azure cli with an active access token, for this authentication method to succeed. Run `az login` to login to azure cli. Run `az account get-access-token` to refresh token if needed.

Reply:
The way our az-cli integration works in kusto-python is that it picks up the cached refresh token and passes it to MSAL.
Once MSAL has that, it can produce new tokens and new refresh tokens on it's own (that's what the refresh token is there for).
If the initial attempt succeeded, the python SDK auth is now independent of az's.

Reply:
Ah okay, I see now. I looked at the src code before but must have missed that. Thank you for filling my gap. Glad it was implemented that way as this is much cleaner:) Approving suggestion.